### PR TITLE
added --lite-help option to trainer for datalab

### PR DESCRIPTION
**Pull request by brandondutra**
For datalab's magic, how about it calls --lite-help?

```
$ python -m trainer.task --lite-help

  --epsilon EPSILON     tf.train.AdamOptimizer epsilon. Only used in dnn
                        models.
  --l1_regularization L1_REGULARIZATION
                        L1 term for linear models.
  --l2_regularization L2_REGULARIZATION
                        L2 term for linear models.
  --model-type {linear_classification,linear_regression,dnn_classification,dnn_regression}
  --top-n TOP_N         For classification problems, the output graph will
                        contain the labels and scores for the top n classes.

Note that if using a DNN model, --hidden-layer-size=NUM, --hidden-layer-
size=NUM, ..., should be used.


```

```
$ python -m trainer.task --help
usage: task.py [-h] [--lite-help] --train-data-paths TRAIN_DATA_PATHS
               --eval-data-paths EVAL_DATA_PATHS --job-dir JOB_DIR
               --analysis-output-dir ANALYSIS_OUTPUT_DIR [--run-transforms]
               [--learning-rate LEARNING_RATE] [--max-steps MAX_STEPS]
               [--num-epochs NUM_EPOCHS] [--train-batch-size TRAIN_BATCH_SIZE]
               [--eval-batch-size EVAL_BATCH_SIZE]
               [--min-eval-frequency MIN_EVAL_FREQUENCY]
               [--save-checkpoints-secs SAVE_CHECKPOINTS_SECS]
               [--epsilon EPSILON] [--l1_regularization L1_REGULARIZATION]
               [--l2_regularization L2_REGULARIZATION] --model-type
               {linear_classification,linear_regression,dnn_classification,dnn_regression}
               [--top-n TOP_N]

optional arguments:
  -h, --help            show this help message and exit
  --lite-help           Show a smaller help message and exit
  --train-data-paths TRAIN_DATA_PATHS
  --eval-data-paths EVAL_DATA_PATHS
  --job-dir JOB_DIR
  --analysis-output-dir ANALYSIS_OUTPUT_DIR
                        Output folder of analysis. Should contain the schema,
                        stats, and vocab files. Path must be on GCS if running
                        cloud training.
  --run-transforms      If used, input data is raw csv that needs
                        transformation.
  --learning-rate LEARNING_RATE
                        optimizer learning rate
  --max-steps MAX_STEPS
                        Maximum number of training steps to perform.
  --num-epochs NUM_EPOCHS
                        Maximum number of training data epochs on which to
                        train. If both --max-steps and --num-epochs are
                        specified, the training job will run for --max-steps
                        or --num-epochs, whichever occurs first. If
                        unspecified will run for --max-steps.
  --train-batch-size TRAIN_BATCH_SIZE
  --eval-batch-size EVAL_BATCH_SIZE
  --min-eval-frequency MIN_EVAL_FREQUENCY
                        Minimum number of training steps between evaluations
  --save-checkpoints-secs SAVE_CHECKPOINTS_SECS
                        How often the model should be checkpointed/saved in
                        seconds
  --epsilon EPSILON     tf.train.AdamOptimizer epsilon. Only used in dnn
                        models.
  --l1_regularization L1_REGULARIZATION
                        L1 term for linear models.
  --l2_regularization L2_REGULARIZATION
                        L2 term for linear models.
  --model-type {linear_classification,linear_regression,dnn_classification,dnn_regression}
  --top-n TOP_N         For classification problems, the output graph will
                        contain the labels and scores for the top n classes.

Note that if using a DNN model, --hidden-layer-size=NUM, --hidden-layer-
size=NUM, ..., should be used.

```

### CODE REVIEW

File: solutionbox/code_free_ml/mltoolbox/code_free_ml/trainer/task.py [hunk 1]
```diff
@@ -58,41 +58,120 @@
...
+    # a shorter help string when --lite-help is used.
+    self.full_parser = argparse.ArgumentParser(epilog=epilog)
+    self.lite_parser = argparse.ArgumentParser(usage='', epilog=epilog, add_help=False)
+
+    # The arguments added here are required to exist by Datalab's magics.
+
+    # Datalab help string
+    self.full_parser.add_argument(
+        '--lite-help', action=self.make_lite_help_action(),
+        help='Show a smaller help message and exit')
```
Comment (qimingj) [inline solutionbox/code_free_ml/mltoolbox/code_free_ml/trainer/task.py]: --lite-help, with the help string, seems indicating it is showing some frequently used arguments. Add "datalab used only" in the help string?

File: solutionbox/code_free_ml/mltoolbox/code_free_ml/trainer/task.py [hunk 2]
```diff
@@ -58,41 +58,120 @@
...
+        '--analysis-output-dir', type=str, required=True,
+        help=('Output folder of analysis. Should contain the schema, stats, and '
+              'vocab files. Path must be on GCS if running cloud training.'))
+    self.full_parser.add_argument(
+        '--run-transforms', action='store_true', default=False,
+        help='If used, input data is raw csv that needs transformation.')
+
+    # HP parameters
+    self.full_parser.add_argument(
+        '--learning-rate', type=float, default=0.01, help='optimizer learning rate')
```
Comment (qimingj) [inline solutionbox/code_free_ml/mltoolbox/code_free_ml/trainer/task.py]: I think learning-rate, max-steps, num-epochs, train-batch-size, eval-batch-size, min-eval-frequency, save-checkpoints-secs also belongs to lite help. They may not apply in another package. Also, in Datalab, the only parameters that will always be passed to command line would be input data, output dir and run-transforms flag. These are required parameters. Everything else is passed through to the package itself.

File: solutionbox/code_free_ml/mltoolbox/code_free_ml/trainer/task.py (hunk 1 above)
Comment (brandondutra) [inline solutionbox/code_free_ml/mltoolbox/code_free_ml/trainer/task.py]: done

File: solutionbox/code_free_ml/mltoolbox/code_free_ml/trainer/task.py (hunk 2 above)
Comment (brandondutra) [inline solutionbox/code_free_ml/mltoolbox/code_free_ml/trainer/task.py]: done. I fully support less things required in datalab

**Pull request merged**